### ISSUE TITLE
[BUG]: Error from non existent dim in QuantileLoss.loss

### ISSUE THREAD

**Issue by bobbyx27** [sktime/pytorch-forecasting#1795]
**WHERE**
In pytorch_forecasting.metrics.quantile
In QuantileLoss.loss (shown below)

**WHAT**
torch.cat with dim=2 does not work as tensors are expected to only have 2 dimensions.
It works fine with dim=1.

(By the way I'm not sure about the criterion's scaling "2 *" logic. I don't find it in other references)

**CODE**
def loss(self, y_pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
    # calculate quantile loss
    losses = []
    for i, q in enumerate(self.quantiles):
        errors = target - y_pred[..., i]
        losses.append(torch.max((q - 1) * errors, q * errors).unsqueeze(-1))
    losses = _2 *_ torch.cat(losses, **dim=2**)

    return losses
